force-app/main/default/classes/ElevateCreateBatchItemRequest.cls

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) 2020, Salesforce.org
+ *     All rights reserved.
+ *
+ *     Redistribution and use in source and binary forms, with or without
+ *     modification, are permitted provided that the following conditions are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of Salesforce.org nor the names of
+ *       its contributors may be used to endorse or promote products derived
+ *       from this software without specific prior written permission.
+ *
+ *     THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *     "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *     LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *     FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *     COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *     INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *     BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *     LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *     CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *     LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *     ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *     POSSIBILITY OF SUCH DAMAGE.
+ */
+public with sharing class ElevateCreateBatchItemRequest {
+    //TODO: If we choose to support ACH we will need to explicitly pass paymentMethodType or the callout will fail
+
+    public ElevateBatchItemType batchItemType;
+    public CommitmentInfo commitmentInfo;
+    public PurchaseInfo purchaseInfo;
+
+    public ElevateCreateBatchItemRequest(ElevateCreateBatchItemRequestDTO batchItemRequestDTO) {
+        if (batchItemRequestDTO.schedule == null || batchItemRequestDTO.schedule.isEmpty()) {
+            batchItemType = ElevateBatchItemType.ONE_TIME;
+        } else {
+            batchItemType = ElevateBatchItemType.COMMITMENT;
+        }
+
+        buildRequestBody(batchItemRequestDTO);
```

Comment:
Do you think we could use the existing `PS_CommitmentRequest` to build out the request body here? I see a lot of similarities in the inner class types we are creating here and I think it is an opportunity for us to be aligned in one class

Reply:
Discussed this one with @voduyemi and we decided that due to the simplicity of being able to use the DTO as a constructor parameter to build the batch item request, that would be the better way to go than implementing a more complex builder pattern for this request since we do not need it right now since the entire request can be built from the data passed from the UI.